GCC 11 output may be truncated copying 15 bytes from a string of length 63

For now here is record of the issue I've run into, I can provide more findings later:
```
In file included from /usr/include/string.h:535,
                 from /home/tksuoran/git/tksuoran/erhe/build/_deps/joltphysics-src/Build/../Jolt/Core/Core.h:385,
                 from /home/tksuoran/git/tksuoran/erhe/build/_deps/joltphysics-src/Build/..//Jolt/Jolt.h:8,
                 from /home/tksuoran/git/tksuoran/erhe/build/_deps/joltphysics-build/CMakeFiles/Jolt.dir/cmake_pch.hxx:,
                 from <command-line>:
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘void JPH::SetThreadName(const char*)’ at /home/tksuoran/git/tksuoran/erhe/build/_deps/joltphysics-src/Jolt/Core/JobSystemThreadPool.cpp:301:10,
    inlined from ‘void JPH::JobSystemThreadPool::ThreadMain(int)’ at /home/tksuoran/git/tksuoran/erhe/build/_deps/joltphysics-src/Jolt/Core/JobSystemThreadPool.cpp:313:15:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:34: error: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output may be truncated copying 15 bytes from a string of length 63 [-Werror=stringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   96 |                                   __glibc_objsize (__dest));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
```
Compiler:
```
$ c++ --version
c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/Docs/ReleaseNotes.md
+++ b/Docs/ReleaseNotes.md
@@ -37,6 +37,7 @@ For breaking API changes see [this document](https://github.com/jrouwe/JoltPhysi
 * When creating a MeshShape with triangles that have near identical positions it was possible that the degenerate check decided that a triangle was not degenerate while the triangle in fact would be degenerate after vertex quantization. The simulation would crash when colliding with this triangle.
 * A scaled compound shape with a center of mass of non zero would not apply the correct transform to its sub shapes when colliding with a soft body
 * A soft body without any edges would hang the solver
+* Fixed GCC 11.4 warning in JobSystemThreadPool.cpp: output may be truncated copying 15 bytes from a string of length 63
 
 ## v4.0.2
 

--- a/Jolt/Core/JobSystemThreadPool.cpp
+++ b/Jolt/Core/JobSystemThreadPool.cpp
@@ -297,16 +297,15 @@ void JobSystemThreadPool::QueueJobs(Job **inJobs, uint inNumJobs)
 #elif defined(JPH_PLATFORM_LINUX)
 	static void SetThreadName(const char *inName)
 	{
-		char truncated_name[16] = { 0 };
-		strncpy(truncated_name, inName, min<size_t>(sizeof(truncated_name), 15));
-		prctl(PR_SET_NAME, truncated_name, 0, 0, 0);
+		JPH_ASSERT(strlen(inName) < 16); // String will be truncated if it is longer
+		prctl(PR_SET_NAME, inName, 0, 0, 0);
 	}
 #endif // JPH_PLATFORM_LINUX
 
 void JobSystemThreadPool::ThreadMain(int inThreadIndex)
 {
 	// Name the thread
-	char name[64];
+	char name[16];
 	snprintf(name, sizeof(name), "Worker %d", int(inThreadIndex + 1));
 
 #if defined(JPH_PLATFORM_WINDOWS) || defined(JPH_PLATFORM_LINUX)

